horizontal-picker/src/main/java/com/github/jhonnyx2012/horizontalpicker/HorizontalPickerRecyclerView.java

```diff
@@ -1,149 +1,159 @@
-            switch (newState){
-                case RecyclerView.SCROLL_STATE_IDLE:
-                    listener.onStopDraggingPicker();
-                    int position = (int) ((computeHorizontalScrollOffset()/itemWidth)+3.5);
-                    if(position!=-1&&position!=lastPosition)
-                    {
-                        selectItem(true,position);
-                        selectItem(false,lastPosition);
-                        lastPosition=position;
-                    }
-                    break;
-                case SCROLL_STATE_DRAGGING:
-                    listener.onDraggingPicker();
-            break;
-            }
-        }
-
-        @Override
-        public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
-            super.onScrolled(recyclerView, dx, dy);
-        }
-    };
-
-    private void selectItem(boolean isSelected,int position) {
-        adapter.getItem(position).setSelected(isSelected);
-        adapter.notifyItemChanged(position);
-        if(isSelected)
-        {
-            listener.onDateSelected(adapter.getItem(position));
-        }
+  private void selectItem(int position) {
+    if (position == NO_POSITION) {
+      return;
     }
-
-    public void setListener(HorizontalPickerListener listener) {
-        this.listener = listener;
+    if (position != adapter.getSelectedPosition()) {
+      int unselectPosition = adapter.getSelectedPosition();
+      adapter.setSelectedPosition(position);
+      adapter.notifyItemChanged(unselectPosition);
+      adapter.notifyItemChanged(position);
+      listener.onDateSelected(adapter.getItem(position));
     }
-
-    @Override
-    public void onClickView(View v, int adapterPosition) {
-        if(adapterPosition!=lastPosition)
-        {
-            selectItem(true,adapterPosition);
-            selectItem(false,lastPosition);
-            lastPosition=adapterPosition;
-        }
+  }
+
+  public void setListener(HorizontalPickerListener listener) {
+    this.listener = listener;
+  }
+
+  @Override
+  public void onClickView(View v, int adapterPosition) {
+    if (adapterPosition != adapter.getSelectedPosition()
+        && adapter.getItemViewType(adapterPosition) != R.layout.item_placeholder) {
+      selectItem(adapterPosition);
+      if (scrollToSelectedPosition) {
+        smoothScrollToPosition(adapterPosition);
+      }
     }
-
-    @Override
-    public void onClick(View v) {
-        setDate(new DateTime());
+  }
+
+  @Override
+  public void onClick(View v) {
+
+  }
+
+  @Override
+  public void smoothScrollToPosition(int position) {
+    final RecyclerView.SmoothScroller smoothScroller = new CenterSmoothScroller(getContext());
+    smoothScroller.setTargetPosition(position);
+    post(new Runnable() {
+      @Override
+      public void run() {
+        layoutManager.startSmoothScroll(smoothScroller);
+      }
+    });
+  }
+
+  public void setDate(LocalDate date) {
+    Day firstDay = adapter.getItem(OFFSET);
```

Comment:
could you add more detail about which each `offset` is?

Reply:
Yep, this is the number of "dummy" views that we need to place in the adapter at the beginning of the list and at the end of the list in order to have the following behavior:

![screen shot 2018-10-02 at 4 26 07 pm](https://user-images.githubusercontent.com/447889/46382483-03efb000-c660-11e8-9881-adbcfbfe07b4.png)

Added a comment in the code